src/interface/variable.hpp

```diff
@@ -86,7 +86,7 @@ class CellVariable {
   bool IsSet(const MetadataFlag bit) const { return m_.IsSet(bit); }
 
   ParArrayND<T> data;
-  ParArrayND<T> flux[3];  // used for boundary calculation
+  ParArrayND<T> flux[4];  // used for boundary calculation
```

Comment:
as with variable.cpp, why not use named coordinate X3DIR here?

Reply:
this one is the array size.  I could do X3DIR+1?

Reply:
Maybe along with `X1DIR`, etc., also define `NDIM` and use it here?